dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4381,6 +4381,420 @@ class FunctionRepeat : public IFunction
+        const ColumnString::Offsets & col_vector_str_offsets,
+        const ColumnString::Chars_t & col_vector_substr_value,
+        const ColumnString::Offsets & col_vector_substr_offsets,
+        size_t val_num,
+        ColumnInt64::Container & res_vec)
+    {
+        ColumnString::Offset prev_offset = 0;
+        const ColumnString::Offset substr_start_offset = 0;
+        auto substr_end_offset = col_vector_substr_offsets[0];
+        for (size_t row = 0; row < val_num; ++row)
+        {
+            res_vec[row] = find(col_vector_str_value, prev_offset, col_vector_str_offsets[row], col_vector_substr_value, substr_start_offset, substr_end_offset);
+
+            prev_offset = col_vector_str_offsets[row];
+        }
+    }
+
+    static void vectorVector(
+        const ColumnString::Chars_t & col_vector_str_value,
+        const ColumnString::Offsets & col_vector_str_offsets,
+        const ColumnString::Chars_t & col_vector_substr_value,
+        const ColumnString::Offsets & col_vector_substr_offsets,
+        size_t val_num,
+        ColumnInt64::Container & res_vec)
+    {
+        ColumnString::Offset str_prev_offset = 0;
+        ColumnString::Offset substr_prev_offset = 0;
+
+        for (size_t row = 0; row < val_num; ++row)
+        {
+            res_vec[row] = find(col_vector_str_value, str_prev_offset, col_vector_str_offsets[row], col_vector_substr_value, substr_prev_offset, col_vector_substr_offsets[row]);
+
+            str_prev_offset = col_vector_str_offsets[row];
+            substr_prev_offset = col_vector_substr_offsets[row];
+        }
+    }
+};
+
+class FunctionInstr : public IFunction
+{
+public:
+    static constexpr auto name = "instr";
+
+    FunctionInstr() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionInstr>();
+    }
+
+    std::string getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 2; }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        auto first_argument = removeNullable(arguments[0]);
+        if (!first_argument->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of first argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        auto second_argument = removeNullable(arguments[1]);
+        if (!second_argument->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of first argument of function {}", arguments[1]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        auto return_type = std::make_shared<DataTypeInt64>();
+        return (arguments[0]->isNullable() || arguments[1]->isNullable()) ? makeNullable(return_type) : return_type;
```

Comment:
for useDefaultImplementationForNulls() const { return true; }, we don't need to consider nullable
```suggestion
        auto first_argument = arguments[0];
        if (!first_argument->isString())
            throw Exception(
                fmt::format("Illegal type {} of first argument of function {}", arguments[0]->getName(), getName()),
                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
        auto second_argument = arguments[1];
        if (!second_argument->isString())
            throw Exception(
                fmt::format("Illegal type {} of first argument of function {}", arguments[1]->getName(), getName()),
                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);

        return std::make_shared<DataTypeInt64>();
```